lib/repl.js

```diff
@@ -1507,72 +1508,60 @@ function regexpEscape(s) {
 
 // If the error is that we've unexpectedly ended the input,
 // then let the user try to recover by adding more input.
+// Note: `e` (the original exception is not used by the current implemention,
+// but may be needed in the future.)
 function isRecoverableError(e, code) {
-  if (e && e.name === 'SyntaxError') {
-    var message = e.message;
-    if (message === 'Unterminated template literal' ||
-        message === 'Unexpected end of input') {
-      return true;
-    }
+  let recoverable = false;
+
+  // Determine if the point of the any error raised is at the end of the input.
+  // There are two cases to consider:
+  //   1.  Any error raised after we have encountered the 'eof' token.
+  //       This prevents us from declaring partial tokens (like '2e') as
+  //       recoverable.
+  //   2.  Three cases where tokens can legally span lines.  This is
+  //       template, comment, and strings with a backslash at the end of
+  //       the line, indicating a continuation.
+  acorn.plugins.repl = (parser) => {
+    parser.extend('nextToken', (nextToken) => {
+      return function(pos, message) {
+        nextToken.call(this, pos, message);
+
+        if (parser.type === acorn.tokTypes.eof) recoverable = true;
+      };
+    });
 
-    if (message === 'missing ) after argument list') {
-      const frames = e.stack.split(/\r?\n/);
-      const pos = frames.findIndex((f) => f.match(/^\s*\^+$/));
-      return pos > 0 && frames[pos - 1].length === frames[pos].length;
-    }
+    parser.extend('raise', (raise) => {
+      return function(pos, message) {
+        switch (message) {
+          case 'Unterminated template':
+          case 'Unterminated comment':
```

Comment:
👆 Might put a note that these message strings are subject to change and to periodically check in on them.

Reply:
I'll ponder how to word that.  Meanwhile, I'll note that should these message strings change, a test will fail.

Reply:
We have to manually update acorn to run into this issue. Having a failed test in such a circumstance would be fine and we'd have to invest that one way or the other. So I guess it is fine to keep it is as.